test/state/bloom.cpp

```diff
@@ -0,0 +1,67 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2023 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "bloom.hpp"
+#include "state.hpp"
+#include <intx/intx.hpp>
+
+namespace evmone::state
+{
+
+namespace
+{
+/// Adds entry to bloom filter data
+/// based on
+/// https://ethereum.github.io/execution-specs/autoapi/ethereum/shanghai/bloom/index.html?highlight=add_to_bloom#add-to-bloom
+inline void add_to_bloom(bytes& bloom, const bytes_view& bloom_entry)
+{
+    assert(bloom.size() == 256);
+    auto hash = keccak256(bloom_entry);
+
+    for (auto i : {0, 2, 4})
+    {
+        auto bit_to_set = ((hash.bytes[i] & 0x07) << 8) + (hash.bytes[i + 1] & 0xFF);
+        auto bit_index = 0x07FF - bit_to_set;
+        auto byte_index = static_cast<uint8_t>(bit_index / 8);
+        auto bit_value = static_cast<uint8_t>(1 << (7 - (bit_index % 8)));
+        bloom[byte_index] = bloom[byte_index] | bit_value;
+    }
+}
+
+}  // namespace
+
+void fill_bloom(TransactionReceipt& tx_receipt) noexcept
+{
+    tx_receipt.logs_bloom = bytes(256, '\0');
+
+    for (const auto& log : tx_receipt.logs)
+    {
+        add_to_bloom(tx_receipt.logs_bloom, log.addr);
+        for (const auto& topic : log.topics)
+            add_to_bloom(tx_receipt.logs_bloom, topic);
+    }
+}
+
+void combine_blooms(bytes& b_ret, const bytes_view& b) noexcept
+{
+    assert(b_ret.size() == b.size());
+    assert(b_ret.size() == 256);
+
+    auto b_0 = intx::be::unsafe::load<intx::uint512>(b.data());
```

Comment:
Are you sure using `intx` here is the fastest? I wonder if even doing an xor-loop on the `bytes` accomplishes the same.